web/packages/teleterm/src/ui/DocumentCluster/ActionButtons.tsx

```diff
@@ -180,3 +190,77 @@ async function getDatabaseUsers(appContext: IAppContext, dbUri: DatabaseUri) {
     throw e;
   }
 }
+
+function AppButton(props: {
+  isWebApp: boolean;
+  launchUrl: string;
+  connect(): void;
+  onLaunchUrl(): void;
+}) {
+  const ref = useRef<HTMLButtonElement>();
+  const [isOpen, setIsOpen] = useState(false);
+
+  if (!props.isWebApp) {
+    return (
+      <ButtonBorder size="small" onClick={props.connect}>
+        Set up connection
+      </ButtonBorder>
+    );
+  }
+
+  return (
+    <Flex>
+      <ButtonBorder
+        size="small"
+        forwardedAs="a"
+        href={props.launchUrl}
+        onClick={props.onLaunchUrl}
+        target="_blank"
+        title="Launch app in a browser"
+        css={`
+          border-top-right-radius: 0;
+          border-bottom-right-radius: 0;
+        `}
+      >
+        Launch
+      </ButtonBorder>
+      <ButtonBorder
+        css={`
+          border-left: none;
+          border-top-left-radius: 0;
+          border-bottom-left-radius: 0;
+        `}
+        setRef={ref}
+        px={1}
+        size="small"
+        onClick={() => setIsOpen(true)}
+      >
+        <ChevronDown size="small" color="text.slightlyMuted" />
+      </ButtonBorder>
+      <Menu
+        anchorEl={ref.current}
+        open={isOpen}
+        onClose={() => setIsOpen(false)}
+        // hack to properly position the menu
+        getContentAnchorEl={null}
+        anchorOrigin={{
+          vertical: 'bottom',
+          horizontal: 'right',
+        }}
+        transformOrigin={{
+          vertical: 'top',
+          horizontal: 'right',
+        }}
+      >
+        <MenuItem
+          onClick={() => {
+            setIsOpen(false);
+            props.connect();
+          }}
+        >
+          Set up connection
```

Comment:
This can remain as "Set up connection" IMHO. As we discussed over DMs, "Connect" would look weird here.